temporalio/lib/temporalio/cancellation.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'temporalio/error'
+
+module Temporalio
+  # Cancellation representation, often known as a "cancellation token". This is used by clients, activities, and
+  # workflows to represent cancellation in a thread/fiber-safe way.
+  class Cancellation
```

Comment:
Any reason not to just call this CancellationToken?

Reply:
No strong reason. Usually Ruby isn't as wordy as .NET and some people may have familiarity with https://ruby-concurrency.github.io/concurrent-ruby/master/Concurrent/Cancellation.html which this is similar to (though ours is more feature rich). I can change to calling it a token everywhere if we want, though while in theory it may seem confusing to call this "a cancellation" when "a cancellation" also refers to an event, I haven't been tripped up on that in practice documenting it.